app/utils/image_cache_manager.js

```diff
@@ -27,33 +27,33 @@ export default class ImageCacheManager {
             listener(path);
         } else {
             addListener(uri, listener);
-            if (!uri.startsWith('http')) {
+            if (uri.startsWith('http')) {
```

Comment:
This is one problem with our return-early pattern since it is easy to miss important stuff like unsubscribing in the case that the file is already cached